sql/hive/src/test/scala/org/apache/spark/sql/sources/BucketedWriteSuite.scala

```diff
@@ -183,4 +186,25 @@ class BucketedWriteSuite extends QueryTest with SQLTestUtils with TestHiveSingle
       }
     }
   }
+
+  test("write bucketed data with bucketing disabled") {
+    withSQLConf(SQLConf.BUCKETING_ENABLED.key -> "false") {
```

Comment:
hmmm, will a user manually set `BUCKETING_ENABLED` to false and then manually specify bucket information while writing data? cc @rxin @marmbrus